battle-engine.js

```diff
@@ -1633,6 +1633,14 @@ Battle = (function () {
 		if (source === undefined && this.event && this.event.target) source = this.event.target;
 
 		if (this.weather === status.id && this.gen > 2) return false;
+		if (status.id && !this.runEvent('SetWeather', source, source, status)) {
```

Comment:
The current `onSetWeather` handlers in the primal-weather data only checks for the weather being set and returns `false` if the weather shouldn't be set. Are you saying that I should feed `sourceEffect` to the event as well and have the primal-weathers do the error messages?